x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/crossclusteraccess/CrossClusterAccessHeadersForCcsRestIT.java

```diff
@@ -705,6 +713,44 @@ public void testApiKeyCrossClusterAccessHeadersSentSingleRemote() throws Excepti
         }
     }
 
+    public void testRemoteClustersXPackUsage() throws IOException {
+        try (MockTransportService remoteTransport = startTransport("remoteNodeA", threadPool, ConcurrentCollections.newBlockingQueue())) {
+            final TransportAddress transportPortAddress = remoteTransport.getOriginalTransport().boundAddress().publishAddress();
+            final TransportAddress remoteClusterServerPortAddress = remoteTransport.getOriginalTransport()
+                .boundRemoteIngressAddress()
+                .publishAddress();
+            final int numberOfRemoteClusters = randomIntBetween(3, 5);
```

Comment:
I thought it was wasteful to not use the keystore secrets to configure remote clusters with the new model. But I see you point. I updated to randomize for 0 to 5 remote clusters (instead of 2 to 5).